src/main/java/com/sap/oss/phosphor/fosstars/data/github/BanditDataProvider.java

```diff
@@ -10,31 +10,36 @@
 import com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures;
 import com.sap.oss.phosphor.fosstars.model.subject.oss.GitHubProject;
 import com.sap.oss.phosphor.fosstars.model.value.ValueHashSet;
-import com.sap.oss.phosphor.fosstars.util.Yaml;
 import java.io.IOException;
-import java.io.InputStream;
-import java.nio.file.Files;
-import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.Map;
-import java.util.Optional;
+import java.util.function.Predicate;
 import java.util.regex.Pattern;
-import org.apache.commons.collections4.IteratorUtils;
 
 /**
  * The data provider gathers info about how a project uses Bandit for static analysis. In
  * particular, it tries to fill out the following features:
  * <ul>
- *   <li>{@link OssFeatures#RUNS_BANDIT_SCANS}</li>
- *   <li>{@link OssFeatures#USES_BANDIT_SCAN_CHECKS}</li>
+ *  <li>{@link OssFeatures#RUNS_BANDIT_SCANS}</li>
+ *  <li>{@link OssFeatures#USES_BANDIT_SCAN_CHECKS}</li>
  * </ul>
  */
 public class BanditDataProvider extends AbstractStaticScanToolsDataProvider {
 
   /**
-   * A step in a GitHub action that triggers analysis with Bandit.
+   * A Predicate to check the any step in a GitHub action that triggers analysis with Bandit.
    */
-  private static final Pattern RUN_STEP_BANDIT_REGEX_PATTERN
```

Comment:
Isn't it nice to keep this and use it in line 38 and 40 rather than creating Pattern twice.

Reply:
There is a slight difference in these regular patterns, the key to search for in the step and then matching the value of the key. But again you are right about the redundancy, i did not find a better way to integrate one predicate and check for everything.

Reply:
@ManjunathMS35 lets fix this here #873 ?